Gems/AzQtComponentsForPython/Code/Platform/Linux/azqtcomponentsforpython_file_settings.cmake

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+# Find the paths for the pyside2/shiboken modules.
+ly_get_package_association("pyside2" PYSIDE_PACKAGE_NAME)
+ly_package_get_target_folder(${PYSIDE_PACKAGE_NAME} PYSIDE2_DOWNLOAD_LOCATION)
+set(PYSIDE2_MODULE_DIR ${PYSIDE2_DOWNLOAD_LOCATION}/${PYSIDE_PACKAGE_NAME})
+
+# Find the qt package location.
+ly_get_package_association("Qt" QT_PACKAGE_NAME)
+ly_package_get_target_folder(${QT_PACKAGE_NAME} QT_DOWNLOAD_LOCATION)
+set(QT_MODULE_DIR ${QT_DOWNLOAD_LOCATION}/${QT_PACKAGE_NAME}/qt)
+
+set(PYSIDE2_PATH "${PYSIDE2_MODULE_DIR}/pyside2")
+set(INCLUDE_DIR "${PYSIDE2_PATH}/include")
+set(PYSIDE2_SHARE_PATH "${PYSIDE2_PATH}/share/PySide2")
+set(PYSIDE2_TYPESYSTEM_PATH "${PYSIDE2_SHARE_PATH}/typesystems")
+set(PYSIDE2_BIN_PATH "${PYSIDE2_PATH}/bin")
+set(PYSIDE2_INCLUDE_DIR ${INCLUDE_DIR}/PySide2)
+set(SHIBOKEN_GENERATOR_EXE_PATH "${PYSIDE2_BIN_PATH}/shiboken2${CMAKE_EXECUTABLE_SUFFIX}") 
+ 
+set(SHIBOKEN_MODULE_PATH "${PYSIDE2_MODULE_DIR}/pyside2/lib")
+set(PYSIDE2_LIB_DIR ${PYSIDE2_PATH}/lib)  
+
+set(SHIBOKEN_INCLUDE_DIR "${INCLUDE_DIR}/shiboken2")
+
+set(SHIBOKEN_SHARED_LIBRARIES "${SHIBOKEN_MODULE_PATH}/libshiboken2.abi3.so")
+set(SHIBOKEN_SHARED_LIBRARIES_D "${SHIBOKEN_MODULE_PATH}/libshiboken2.abi3.so")
+set(PYSIDE2_SHARED_LIBRARIES "${PYSIDE2_LIB_DIR}/libpyside2.abi3.so")
+set(PYSIDE2_SHARED_LIBRARIES_D "${PYSIDE2_LIB_DIR}/libpyside2.abi3.so")
```

Comment:
It feels like all of this should have been inside our pyside2 or qt package itself, and should be done via import targets instead of paths (due to the way that interacts with installer and deployment and config autogeneration).   

It might be a bit of a pain to update those 2 packages to conform though.  In which case at least we could use imported targets here?  If you need help with how imported targets work, please let me know.  its just a slightly different syntax for a similar result but is more platform/config friendly

Reply:
yeah I looked at our findpyside2.cmake in our pyside2 package and one of these days we're going to have to rev that and when we do, we should move all of this kind of stuff in there and refactor it.

For now, this kinda will work

Reply:
Yeah, I think it will work here even in installer mode because we are already "anchoring" these paths to an installed 3p package, so it will still work even in an installer.  If we were installing our own custom files it would not function.  an example of how you might express this in targets, in case its helpful.

```

# you can refer to Shiboken::Lib if you want to get its include dir, shared libraries, etc.
add_target(Shiboken::Lib IMPORTED GLOBAL)
ly_target_include_system_directories(TARGET Shiboken::Lib INTERFACE ${SHIBOKEN_INCLUDE_DIR})
set_target_properties(Shiboken::Lib PROPERTIES IMPORTED_LOCATION ${SHIBOKEN_SHARED_LIBRARIES})
set_target_properties(Shiboken::Lib PROPERTIES IMPORTED_LOCATION_DEBUG ${SHIBOKEN_SHARED_LIBRARIES_D})

# refer to pyside2::Lib to get its include dir, shared libraries, et.c
add_target(pyside2::Lib IMPORTED GLOBAL)
ly_target_include_system_directories(TARGET pyside2::Lib INTERFACE ${PYSIDE2_INCLUDE_DIR})
set_target_properties(pyside2::Lib PROPERTIES IMPORTED_LOCATION ${PYSIDE2_SHARED_LIBRARIES})
set_target_properties(pyside2::Lib PROPERTIES IMPORTED_LOCATION_DEBUG ${PYSIDE2_SHARED_LIBRARIES_D})

# you can refer to Shiboken::Tool instead of an exe path, inside custom_commands and targets.
add_executable(Shiboken::Tool IMPORTED GLOBAL)
set_target_properties(Shiboken::Tool PROPERTIES IMPORTED_LOCATION "${SHIBOKEN_GENERATOR_EXE_PATH}")
```

However, likely we'd want to (next time we rev the pyside2 3p package, put these declarations inside the real pyside2 3p package so that you can just find_package it and just use them...